frontend/src/components/IstioWizards/GatewaySelector.tsx

```diff
@@ -57,10 +60,33 @@ export class GatewaySelector extends React.Component<Props, GatewaySelectorState
       selectedGateway: props.gateways.length > 0 ? (props.gateway !== '' ? props.gateway : props.gateways[0]) : '',
       gatewayClass: '',
       addMesh: props.isMesh,
-      port: 80
+      port: 80,
+      isOpen: false,
     };
   }
 
+  onToggleClick = () => {
+    this.setState((prevState) => ({ isOpen: !prevState.isOpen }));
+  };
+
+  onSelect = (_event: React.MouseEvent<Element, MouseEvent> | undefined, value: string | number | undefined) => {
+    this.setState({ selectedGateway: value as string, isOpen: false });
+  };
+
+  toggle = (toggleRef: React.Ref<any>) => (
+    <MenuToggle
+      ref={toggleRef}
+      onClick={this.onToggleClick}
+      isExpanded={this.state.isOpen} 
+      isDisabled={!this.state.addGateway || this.state.newGateway || this.props.gateways.length === 0} 
+      style={{
```

Comment:
I think it is preferable to remove the fix width, so the select component adjusts to the width of the content

Reply:
resolved